tests/mocks/pocketjs.ts

```diff
@@ -0,0 +1,265 @@
+  fail = false
+  rpcMockError = new RpcError('500', 'Mock error')
+  relayRequest = '{"method":"eth_blockNumber","params":[],"id":1,"jsonrpc":"2.0"}'
+  relayResponse: string | RelayResponse | RpcError = '{"id":1,"jsonrpc":"2.0","result":"0x1083d57"}'
+  session = DEFAULT_MOCK_VALUES.SESSION
+
+  constructor(
+    // TODO: Implement usage with custom session generation
+    // { randomizeNodes = false }: PocketMockOptions,
+    dispatchers?: URL[],
+    rpcProvider?: HttpRpcProvider,
+    configuration?: Configuration
+  ) {
+    // Cannot set rpcProvider without setting the dispatchers as it depends on them
+    if (rpcProvider && !dispatchers) {
+      throw new Error('Cannot set rpcProvider without dispatchers')
+    }
+
+    this.dispatchers = dispatchers || DEFAULT_MOCK_VALUES.DISPATCHERS
+    this.rpcProvider = rpcProvider || new HttpRpcProvider(DEFAULT_MOCK_VALUES.DISPATCHERS[0])
+    this.configuration = configuration || DEFAULT_MOCK_VALUES.CONFIGURATION
+  }
+
+  _setMockOptions({ randomizeNodes }: PocketMockOptions): void {
+    if (randomizeNodes) {
+      const nodes = PocketMock._genRandomNodes()
+
+      this.nodes = nodes
+    } else {
+      this.nodes = DEFAULT_MOCK_VALUES.NODES
+    }
+  }
+
+  // Generate a pocket session's amount of nodes (5) with random field values
+  static _genRandomNodes(): Node[] {
+    const generateString = (length: number) =>
+      Array.from(Array(length), () => Math.floor(Math.random() * 36).toString(36)).join('')
+
+    const nodes: [Node, Node, Node, Node, Node] = [<Node>{}, <Node>{}, <Node>{}, <Node>{}, <Node>{}]
+
+    nodes.forEach(
+      () =>
+        new Node(
+          generateString(40),
+          generateString(64),
+          false,
+          StakingStatus.Staked,
+          BigInt(Math.floor(Math.random() * MINIMUM_NODE_STAKE) + 10000),
+          `https://validators${Math.floor(Math.random() * 20)}.com`,
+          // Chain format: ['0000', '00XX']
+          new Array(4).fill('').map(() => `00${Math.floor(Math.random() * 20)}`.padStart(2, '0'))
+        )
+    )
+    return nodes
+  }
+
+  /**
+   * GetObject returns an object instance of pocketjs  with the configured options
+   */
+  async getObject(): Promise<Pocket> {
+    const sessionManagerMock = new Mock<SessionManager>()
+      .setup((instance) => instance.getCurrentSession(It.IsAny(), It.IsAny(), It.IsAny()))
+      .returnsAsync(this.session)
+      .object()
+
+    const response = this._getRelayResponse()
+
+    const repoMock = new Mock<Pocket>()
+      .setup((instance) => instance.sessionManager)
+      .returns(sessionManagerMock)
+      .setup((instance) =>
+        instance.sendRelay(
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny(),
+          It.IsAny()
+        )
+      )
+      .returnsAsync(response)
```

Comment:
We should look into defining fixture files for sync-checker responses and dynamically use code like this to simulate sync response from every type of network we have, ETH, BSC, yatatata.

Reply:
completely agree, right now this is an initial implementation but is flexible enough to cater custom needs